samples/e2eTestUtils/src/CertificateUtils.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const { X509Certificate, createPrivateKey } = require("crypto");
+const path = require("path");
+const fs = require("fs");
+const { execSync } = require("child_process");
+
+import { getKeyVaultSecret } from "./KeyVaultUtils";
+
+// define paths for temporary files
+const p12FilePath = path.join(__dirname, "certificate.p12");
+const certificateKEY = path.join(__dirname, "certificate.key");
+const certificateCER = path.join(__dirname, "certificate.cer");
+
+export const getCertificateInfo = async (
+    client: any,
+    secretName: string
+): Promise<Array<string>> => {
+    const PKCS12CertificateBase64: string = await getKeyVaultSecret(
+        client,
+        secretName
+    );
+
+    // get the private key and the public certificate, in PKCS 12 format
+    const pkcs12Certificate = Buffer.from(PKCS12CertificateBase64, "base64");
+
+    // write the PKCS#12 certificate to a temporary file
+    fs.writeFileSync(p12FilePath, pkcs12Certificate);
+
+    try {
+        // get the private key from the pkcs12 file through openssl, via a synchronous child process
+        execSync(
+            `openssl pkcs12 -in ${p12FilePath} -nocerts -nodes -passin pass: | sed -ne '/-BEGIN PRIVATE KEY-/,/-END PRIVATE KEY-/p' > ${certificateKEY}`
```

Comment:
Per @gladjohn: are there any node js libraries like pkcs12, or pem to handle PKCS#12 files directly in memory? why use openssl?

Reply:
@gladjohn this functionality doesn't exist natively in NodeJS. We want to avoid pulling in 3rd party packages. Hence, using NodeJS's native child_process to run openssl.